01_fib/CameronSchafer/fib.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+//base fibonacci function
+func fib(n int) {
+	sign, n1, n2 := 1, 0, 1
+	//check if use nega or normal fib sequence
+	if n < 0 {
+		sign = -1
+	}
+	//loop until all numbers have been printed
+	for i := 0; i < n*sign; i++ {
+		if i%2 == 0 || sign > 0 {
+			//will get the negafib number if the sign is -1
+			n1, n2 = n2, n1+n2*sign
```

Comment:
For my own benefit, can you tell me does this evaluate everything on the right of `=` prior to assignment?  I guess it must.

Reply:
Thanks I found it does do that, this has helped me simplify my second lab 👍

Reply:
Yes it does. This assignment is shorthand for
```
n1 = n2
n2 = n1 + (n2 * sign)
```

Reply:
From my experiments it is more closely equivalent to

```
x1 := n1
x2 := n2

n1 = x2 
n2 = x1 + (x2 * sign)
```
So the values on the right hand side are always the original values before any assignment occurs.

So you can swap variable values like this

```
n1, n2 = n2, n1
```

You can see if it translated as seperate statements you would end up with the same value in `n1` and `n2`

Reply:
Yeah that makes sense, it is a really useful feature of Go.